tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetryCollector.cs

```diff
@@ -0,0 +1,175 @@
+
+            _azureApServicesMetadata = appServicesMetadata;
+            var host = HostMetadata.Instance;
+            _hostData = new HostTelemetryData
+            {
+                ContainerId = ContainerMetadata.GetContainerId(),
+                Os = FrameworkDescription.Instance.OSPlatform,
+                OsVersion = Environment.OSVersion.ToString(),
+                Hostname = host.Hostname,
+                KernelName = host.KernelName,
+                KernelRelease = host.KernelRelease,
+                KernelVersion = host.KernelVersion,
+            };
+
+            _isTracerInitialized = true;
+            SetHasChanges();
+        }
+
+        public void RecordSecuritySettings(SecuritySettings securitySettings)
+        {
+            _securitySettings = securitySettings;
+            SetHasChanges();
+        }
+
+        public bool HasChanges()
+        {
+            return _isTracerInitialized && _hasChangesFlag == 1;
+        }
+
+        /// <summary>
+        /// Get the application data. Will be null if not yet initialized.
+        /// </summary>
+        public ApplicationTelemetryData GetApplicationData()
+        {
+            var settings = _settings;
+            return settings?.ApplicationData;
+        }
+
+        /// <summary>
+        /// Get the application data. Will be null if not yet initialized.
+        /// </summary>
+        public HostTelemetryData GetHostData()
+        {
+            return _hostData;
+        }
+
+        /// <summary>
+        /// Get the latest data to send to the intake.
+        /// </summary>
+        /// <returns>Null if there are no changes, or the collector is not yet initialized</returns>
+        public ConfigTelemetryData GetConfigurationData()
+        {
+            var hasChanges = Interlocked.CompareExchange(ref _hasChangesFlag, 0, 1) == 1;
+            if (!_isTracerInitialized || !hasChanges)
+            {
+                return null;
+            }
+
+            var settings = _settings.Settings;
+
+            var data = new ConfigTelemetryData
+            {
+                Platform = FrameworkDescription.Instance.ProcessArchitecture,
+                Enabled = settings.TraceEnabled,
+                AgentUrl = settings.Exporter.AgentUri.ToString(),
+                Debug = GlobalSettings.Source.DebugEnabled,
+#pragma warning disable CS0618
+                AnalyticsEnabled = settings.AnalyticsEnabled,
+#pragma warning restore CS0618
+                SampleRate = settings.GlobalSamplingRate,
+                SamplingRules = settings.CustomSamplingRules,
+                LogInjectionEnabled = settings.LogsInjectionEnabled,
+                RuntimeMetricsEnabled = settings.RuntimeMetricsEnabled,
+                RoutetemplateResourcenamesEnabled = settings.RouteTemplateResourceNamesEnabled,
+                PartialflushEnabled = settings.Exporter.PartialFlushEnabled,
+                PartialflushMinspans = settings.Exporter.PartialFlushMinSpans,
```

Comment:
These properties could also have matching casing
```suggestion
                PartialFlushEnabled = settings.Exporter.PartialFlushEnabled,
                PartialFlushMinSpans = settings.Exporter.PartialFlushMinSpans,
```

Reply:
Just like the above example, it looks like the original casing you had would result in JSON keys `partialflush_enabled` and `partialflush_minspans` which seems good to me 👍🏼